include/triton/Analysis/Allocation.h

```diff
@@ -18,16 +18,39 @@ namespace mlir {
 namespace triton {
 class AllocationAnalysis;
 
-SmallVector<unsigned>
-getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
-                             unsigned &outVec);
-SmallVector<unsigned> getScratchConfigForCvtLayout(RankedTensorType srcType,
-                                                   RankedTensorType dstType,
-                                                   unsigned &inVec,
-                                                   unsigned &outVec);
-SmallVector<unsigned> getRepShapeForCvtLayout(triton::gpu::ConvertLayoutOp op);
-SmallVector<unsigned> getRepShapeForCvtLayout(RankedTensorType srcTy,
-                                              RankedTensorType dstTy);
+struct ScratchConfig {
+  SmallVector<unsigned> repShape;
+  SmallVector<unsigned> paddedRepShape;
+  unsigned inVec{1};
+  unsigned outVec{1};
+  unsigned paddedDim;
+  unsigned paddedSize{0};
+  unsigned paddedStride;
+
+  ScratchConfig() = default;
+  ScratchConfig(SmallVector<unsigned> repShape,
+                SmallVector<unsigned> paddedRepShape, unsigned inVec,
+                unsigned outVec, unsigned paddedDim, unsigned paddedSize,
+                unsigned paddedStride)
+      : repShape(repShape), paddedRepShape(paddedRepShape), inVec(inVec),
+        outVec(outVec), paddedDim(paddedDim), paddedSize(paddedSize),
+        paddedStride(paddedStride) {}
+
+  void print(llvm::raw_ostream &os) const {
+    os << "repShape: [";
+    llvm::interleaveComma(repShape, os);
+    os << "]";
+    os << ", paddedRepShape: [";
+    llvm::interleaveComma(paddedRepShape, os);
+    os << "]";
+    os << ", inVec: " << inVec << ", outVec: " << outVec
+       << ", paddedDim: " << paddedDim << ", paddedSize: " << paddedSize
+       << ", paddedStride: " << paddedStride << "\n";
+  }
+
+  static ScratchConfig get(RankedTensorType srcTy, RankedTensorType dstTy);
+  static ScratchConfig get(triton::gpu::ConvertLayoutOp op);
```

Comment:
I'm confused by having both a constructor and static `get()` factory functions.  It's not clear what the difference is supposed to be.

A simple change is to make these free functions. `getScratchConfigForConversion(srcTy, dstTy)` is clear.  I might just get rid of the one which takes a ConvertLayoutOp because it's the same as getScratchConfigForConversion(op.getSrcTy(), op.getDstTy()).